tests/Stoa.test.ts

```diff
@@ -324,48 +298,42 @@ describe ('Test of Stoa API Server', () =>
 
     });
 
-    it ('Test of the path /transaction_received', (doneIt: () => void) =>
+    describe ('Test of pending transactions', () =>
     {
-        let uri = URI(host)
-            .port(port)
-            .directory("transaction_received");
-
-        let url = uri.toString();
-        assert.doesNotThrow(async () =>
+        it ('Store one pending transaction', (doneIt: () => void) =>
         {
-            const block = Block.reviver("", sample_data2);
-            await client.post(url, {transaction: block.txs[0]})
+            let uri = URI(host)
+                .port(port)
+                .filename("transaction_received");
 
-            setTimeout(() =>
+            let url = uri.toString();
+            assert.doesNotThrow(async () =>
             {
-                doneIt();
-            }, 100);
+                const block = Block.reviver("", sample_data2);
+                await client.post(url, {transaction: block.txs[0]});
+                setTimeout(doneIt, 100);
+            });
         });
-    });
 
-    it ('Test of the path /pending_transactions/:address', (doneIt: () => void) =>
-    {
-        let uri = URI(host)
-            .port(port)
-            .directory("pending_transactions")
-            .filename("GDAGR22X4IWNEO6FHNY3PYUJDXPUCRCKPNGACETAUVGE3GAWVFPS7VUJ");
-
-        client.get (uri.toString())
-            .then((response) =>
-            {
-                assert.strictEqual(response.data.length, 1);
-                assert.strictEqual(response.data[0].tx_hash, '0x42febd46e93ace' +
-                    'bfc7f81e7a8b0228c5c4fed42de29bb5b4872b09699c28bb3b29e8dbb' +
-                    'c65eb3a46b60ccb688e8a6d4ffbc341a0d59f7de13d28de2fede5566d');
-                assert.strictEqual(response.data[0].address, 'GCOMMONBGUXXP4RFCYGEF74JDJVPUW2GUENGTKKJECDNO6AGO32CUWGU');
-                assert.strictEqual(response.data[0].amount, '1663400000');
-                assert.strictEqual(response.data[0].fee, '0');
-            })
-            .catch((error) =>
-            {
-                assert.ok(!error, error);
-            })
-            .finally(doneIt);
+        it ('Test of the path /transactions/pending/:address', () =>
+        {
+            let uri = URI(host)
+                .port(port)
+                .directory("transactions/pending")
```

Comment:
How did this work before?

Reply:
In the previous code, we found that the internal code was not normally checked.
However, the modified code addresses this.
The link below provides a detailed description and provides a solution to the same situation as ours.
https://www.testim.io/blog/testing-promises-using-mocha/ 
[Section: Why Mocha Struggles With Async Tasks]